tests/manage/z_cluster/cluster_expansion/test_add_capacity_ms.py

```diff
@@ -0,0 +1,120 @@
+import logging
+import pytest
+
+from ocs_ci.utility.utils import ceph_health_check
+from ocs_ci.helpers import helpers
+from ocs_ci.ocs import constants
+from ocs_ci.ocs.resources.pod import cal_md5sum
+from ocs_ci.ocs.ocp import OCP
+from ocs_ci.framework import config
+from ocs_ci.utility.rosa import configure_managed_service_size, get_managed_service_size
+from ocs_ci.helpers.managed_services import verify_provider_topology
+from ocs_ci.framework.testlib import (
+    ignore_leftovers,
+    ManageTest,
+    ms_consumer_required,
+    tier1,
+)
+
+logger = logging.getLogger(__name__)
+
+
+@tier1
+@ignore_leftovers
+@ms_consumer_required
+class TestAddCapacityMS(ManageTest):
+    """
+    Automates adding variable capacity to the cluster
+    """
+
+    @pytest.fixture(autouse=True)
+    def setup(self):
+        """
+        Prepare pods for the test and add finalizer.
+
+        """
+        self.provider_cluster_index = config.get_provider_index()
+        self.consumer_indexes = config.get_consumer_indexes_list()
+
+    def test_add_capacity_managed_service(
+        self, pvc_factory, pod_factory, teardown_project_factory
+    ):
+        """
+        Test to add variable capacity to the OSD cluster while IOs running
+        """
+        SIZE = {
+            "4": "8",
+            "8": "12",
+            "12": "16",
+            "16": "20",
+            "20": "48",
+            "48": "69",
+            "96": "96",
+        }
+        config.switch_ctx(self.consumer_indexes[0])
```

Comment:
The switch is not necessary as the test has ms_consumer_required marker set.

Reply:
On this test we need to switch between consumer cluster an provider cluster

Reply:
So this test should be run from provider only? In that case there needs to be added marker `skipif_ms_consumer` to ensure that this is executed only on provider and it is not executed again on consumer or `runs_on_provider` to ensure that the initial context is always provider.